### PR TITLE
add some conditions

### PR DESCRIPTION
**Pull request by joel**

### CODE REVIEW

File: lib/zip_dir/unzipper.rb [hunk 1]
```diff
@@ -22,16 +22,38 @@ def unzipped?
...
         end
       end
       @unzipped = true
     end
+
+    private
+
+    def unauthorized_file?(file)
+      file.name.match(/MACOSX|.DS_Store/)
+    end
```
Comment (s12chung) [inline lib/zip_dir/unzipper.rb:44]: I don't think this belongs here, it belongs in finalcloud.

Comment (s12chung) [inline lib/zip_dir/unzipper.rb:44]: Or finalcloud should clean it up

File: lib/zip_dir/unzipper.rb [hunk 2]
```diff
@@ -22,16 +22,38 @@ def unzipped?
...
+
+    def ensure_path_exists file_path
+      # Create directory structure if doesn't exists
+      if file_path.to_s.match(/(?<last_caracter>\/$)/)
```
Comment (s12chung) [inline lib/zip_dir/unzipper.rb:48]: What does this match exactly? A test? :p

Comment (joel) [inline lib/zip_dir/unzipper.rb:48]: `my/path/` instead of `my/path`

File: lib/zip_dir/unzipper.rb (hunk 1 above)
Comment (joel) [inline lib/zip_dir/unzipper.rb:44]: could be but when a build file zip are opened and re-zipped we are this into.

Comment (s12chung) [inline lib/zip_dir/unzipper.rb:44]: or at least add it as an option...... it's a public `gem` lol... this is random unexpected behaviour.

File: lib/zip_dir/unzipper.rb (hunk 2 above)
Comment (s12chung) [inline lib/zip_dir/unzipper.rb:48]: test please.... i have stronger requirements for gems. plus `last_caracter` is misspelled......

**Pull request merged**